File: wrappers/java/src/main/java/org/hyperledger/indy/sdk/anoncreds/ClaimRevokedException.java
package org.hyperledger.indy.sdk.anoncreds;

import org.hyperledger.indy.sdk.ErrorCode;
import org.hyperledger.indy.sdk.IndyException;

/**
 * Exception thrown when attempting to use a claim that has been revoked.
 */
public class ClaimRevokedException extends IndyException
{
	private static final long serialVersionUID = 8269746965241515882L;
	private final static String message = "The claim has been revoked.";

	/**
	 * Initializes a new ClaimRevokedException.
	 */
	public ClaimRevokedException() 
    {
    	super(message, ErrorCode.AnoncredsClaimRevoked.value());
    }
}

File: wrappers/dotnet/indy-sdk-dotnet/AnonCredsApi/ClaimRevokedException.cs
namespace Hyperledger.Indy.AnonCredsApi
{
    /// <summary>
    /// Exception thrown when a claim has been revoked.
    /// </summary>
    public class ClaimRevokedException : IndyException
    {
        const string message = "The claim has been revoked.";

        /// <summary>
        /// Initializes a new ClaimRevokedException.
        /// </summary>
        internal ClaimRevokedException() : base(message, (int)ErrorCode.AnoncredsCredentialRevoked)
        {

        }
    }

}
